extensions/objects/templates/literally-canvas-diagram-editor.component.spec.ts

```diff
@@ -0,0 +1,59 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for the literally canvas diagram editor.
+ */
+
+describe('LiterallyCanvasDiagramEditor', function() {
+  var LCDiagramEditorCtrl = null;
+
+  beforeEach(angular.mock.module('oppia'));
+
+  beforeEach(angular.mock.inject(function($componentController) {
+    LCDiagramEditorCtrl = $componentController('literallyCanvasDiagramEditor');
+    var mockDocument = document.createElement('div');
+    mockDocument.setAttribute('id', 'lc');
+    document.getElementById = jasmine.createSpy(
+      'HTML element').and.returnValue(mockDocument);
+    LCDiagramEditorCtrl.$onInit();
+  }));
+
+  it('should update diagram size', function() {
+    var width = 100;
+    var height = 100;
+    LCDiagramEditorCtrl.diagramWidth = width;
+    LCDiagramEditorCtrl.diagramHeight = height;
+    LCDiagramEditorCtrl.onWidthInputBlur();
+    expect(LCDiagramEditorCtrl.currentDiagramWidth).toBe(100);
+    expect(LCDiagramEditorCtrl.lc.width).toBe(100);
+    LCDiagramEditorCtrl.onHeightInputBlur();
+    expect(LCDiagramEditorCtrl.currentDiagramHeight).toBe(100);
+    expect(LCDiagramEditorCtrl.lc.height).toBe(100);
+  });
+
+  it('should return information on diagram size', function() {
+    var maxDiagramWidth = 491;
+    var maxDiagramHeight = 551;
+    var helpText = 'This diagram has a maximum dimension of ' +
```

Comment:
Indent continuation lines and introduce a parentheses if you're breaking across lines:

```
var helpText = (
  'This diagram ... +
  '...');
```

Reply:
Done.